Documentation/contributing/testing/scalability.rst

```diff
@@ -0,0 +1,294 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _scalability_testing:
+
+Scalability and Performance Testing
+===================================
+
+Introduction
+~~~~~~~~~~~~
+
+Cilium scalability and performance tests leverage `ClusterLoader2 <CL2_>`_.
+For overview of ClusterLoader2, please refer to the `Readme <CL2_README_>`_ and `Getting Started <CL2_GETTING_STARTED_>`_.
+At a high level, ClusterLoader2 allows for specifying states of the cluster, how to transition between them
+and what metrics to measure during the test run.
+Additionally, It allows for failing the test if the metrics are not within the expected thresholds.
+
+Overview of existing tests
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Tests based on kOps and GCP VMs:
+
+* 100 nodes scale test - ``/scale-100`` `Workflow <SCALE_100_WORKFLOW_>`_ that executes two test scenarios:
+
+    * `Upstream load test <UPSTREAM_LOAD_TEST_>`_
+
+    * `Network policy scale test <NETPOL_SCALE_TEST_>`_
+
+
+* FQDN performance test - ``/fqdn-perf`` `Workflow <FQDN_PERF_WORKFLOW_>`_
+  a simple two-node test that deploys pods with FQDN policies
+  and measures the time it takes to resolve FQDNs from client point of view.
+
+* ClusterMesh scale test - ``/scale-clustermesh`` `Workflow <CLUSTERMESH_WORKFLOW_>`_ leverages
+  a `mock Clustermesh control-plane <CLUSTERMESH_MOCK_>`_ that simulates large deployments of ClusterMesh.
+
+Test based on EKS:
+
+* Egress Gateway scale test - ``/scale-egw``. `Workflow <EGW_WORKFLOW_>`_ tests Egress Gateway on a small cluster,
+  but with synthetically created Endpoints and Nodes to simulate large cluster.
+
+Whenever developing new tests, you should consider if you want to add a test to already existing workflow,
+create a new one or maybe extend the existing test.
+If you are unsure, you can always ask in the ``#sig-scalabilty`` `Slack channel <SLACK_CHANNEL_>`_.
+For example, if you want to run test on a large cluster,
+you might consider adding it as a separate test scenario to already existing 100 nodes scale test
+to reduce cost of CI, as spinning up a new cluster and tearing it down is quite a long process.
```

Comment:
```suggestion
you might consider adding it as a separate test scenario to the already existing 100-nodes scale test
to reduce the cost of CI, because spinning up a new cluster and tearing it down is quite a long process.
```